codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -480,13 +480,15 @@ private void cancel(Throwable cause) {
                 return;
             }
 
-            for (;;) {
-                FlowControlled frame = pendingWriteQueue.poll();
-                if (frame == null) {
-                    break;
-                }
-                writeError(frame, streamError(stream.id(), INTERNAL_ERROR, cause,
-                                              "Stream closed before write could take place"));
+            FlowControlled frame = pendingWriteQueue.poll();
+            if (frame != null) {
+                // Only create exception once and reuse to reduce overhead of filling in the stacktrace.
+                Http2Exception exception = streamError(stream.id(), INTERNAL_ERROR, cause,
```

Comment:
nit: final?

Reply:
I always prefer having final local variables because it's good for readability. It is absolutely sure it will never change :-)